[Feature]Support hudi connector

adzfolc opened the pull request:
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #https://github.com/StarRocks/starrocks/issues/7681

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

adzfolc committed:
1. add hive_partition_refresh_concurrency in Config, to customize num of thread to refresh hive external table partition info
2. use daemon fixed thread pool to refresh hive partition

adzfolc committed:
1. add hive_partition_refresh_concurrency in Config, to customize num of thread to refresh hive external table partition info
2. use daemon fixed thread pool to refresh hive partition

adzfolc committed:
1. add hive_partition_refresh_concurrency in Config, to customize num of thread to refresh hive external table partition info
2. use daemon fixed thread pool to refresh hive partition

adzfolc committed:
1. fix code style

adzfolc committed:
1. fix code style

adzfolc committed:
1. support hudi connector
2. add test for hudi connector
3. remove unused imports

adzfolc committed:
1. support hudi connector
2. add test for hudi connector
3. remove unused imports

adzfolc committed:
1. support hudi connector
2. add test for hudi connector
3. remove unused imports

adzfolc committed:
1. support hudi connector
2. add test for hudi connector
3. remove unused imports

adzfolc committed:
1. support hudi connector
2. add test for hudi connector
3. remove unused imports

adzfolc committed:
1. remove unused parameter

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify tests

adzfolc committed:
1. modify refresh partition to parallel
2. modify hive_partition_refresh_concurrency to System's core num

adzfolc committed:
1. modify refresh partition to parallel
2. modify hive_partition_refresh_concurrency to System's core num

adzfolc commented:
run starrocks_fe_unittest

adzfolc committed:
1. remove comments

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 25 / 150 (16.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|19|00.00%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|1|68|01.47%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|29|10.34%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

adzfolc committed:
1. fix func call

murphyatwork committed:
[Feature] Introduce NestloopJoinOperator in FE (#6909)

starrocks-xupeng committed:
[Feature] support FE integrate with StarMgr (#7129)

1. FE uses StarManager as a module
2. StarManager uses FE to help achieve meta durability and high availability

adzfolc committed:
1. fix func call

adzfolc committed:
1. fix func call

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 49 / 166 (29.52%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|19|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|5|44|11.36%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|20|68|29.41%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

starrocks-xupeng committed:
Revert "[Feature] support FE integrate with StarMgr (#7129)" (#7937)

This reverts commit d765a45c1558c4b51b485e78999dc1b25bffa05c.

adzfolc commented:
run starrocks_fe_unittest

adzfolc commented:
run starrocks_fe_unittest

adzfolc committed:
1. remove unused func

adzfolc committed:
1. remove unused func

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 49 / 163 (30.06%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|5|44|11.36%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|20|68|29.41%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

adzfolc committed:
1. remove error applied code

adzfolc committed:
1. remove error applied code

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/connector/hudi/HudiMetadata.java:
```diff
@@ -0,0 +1,69 @@
+    private HiveMetaCache metaCache = null;
+    private final String resourceName;
+
+    public HudiMetadata(String resourceName) throws DdlException {
```
```suggestion
    public HudiMetadata(String resourceName) {
```

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 46 / 147 (31.29%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|29|10.34%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|20|68|29.41%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -229,23 +233,110 @@ public static Type convertColumnType(String hiveType) throws DdlException {
+    public static Type convertHudiTableColumnType(Schema avroSchema) throws DdlException {
+        Schema.Type columnType = avroSchema.getType();
+        LogicalType logicalType = avroSchema.getLogicalType();
+        int precision = 0;
```
you may define precision and scale at line 282

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:
```diff
@@ -320,7 +329,49 @@ public Table getTable(HiveTableName hiveTableName) {
     private Table loadTable(HiveTableName hiveTableName) throws TException, DdlException {
         org.apache.hadoop.hive.metastore.api.Table hiveTable = client.getTable(hiveTableName);
-        HiveTable table =  HiveMetaStoreTableUtils.convertToSRTable(hiveTable, resourceName);
+        Table table = null;
```
abstract to a method

adzfolc committed:
1. refactor func
2. remove unused code&exception

adzfolc committed:
1. refactor func
2. remove unused code&exception

adzfolc committed:
1. refactor func

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 43 / 147 (29.25%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|31|09.68%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|17|66|25.76%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

adzfolc commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 43 / 147 (29.25%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|31|09.68%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|17|66|25.76%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:
```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
-        HiveTable table =  HiveMetaStoreTableUtils.convertToSRTable(hiveTable, resourceName);
+        Table table = null;
+        if (hiveTable.getSd().getInputFormat().contains("hudi")) {
+            Schema hudiTable = loadHudiTable(hiveTable);
```
hudiTable -> hudiSchema ?

adzfolc replied:
> hudiTable -> hudiSchema ?

done

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:
```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
+
+        Option<String[]> hudiTablePrimaryKey = hudiTableConfig.getRecordKeyFields();
+        if (hudiTablePrimaryKey.isPresent()) {
+            hudiProperties.put("hudi.table.primaryKey", hudiTableConfig.getRecordKeyFieldProp());
```
I think you may use HudiTable.xxx to replace these strings.

adzfolc replied:
> I think you may use HudiTable.xxx to replace these strings.

done

stephen-shelby commented on fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:
```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
     }
 
+    private Schema loadHudiTable(org.apache.hadoop.hive.metastore.api.Table hiveTable) throws DdlException {
+        Map<String, String> hudiProperties = Maps.newHashMap();
```
It is better to put loadHudiTable and HudiTable.validate together

adzfolc replied:
> It is better to put loadHudiTable and HudiTable.validate together

done

stephen-shelby replied:
I mean loadHudiSchema and HudiTable.validate are duplicate.

adzfolc committed:
1. rename some parameters
2. merge loadHudiTable and HudiTable.validate together

adzfolc committed:
1. change used func

adzfolc commented on fe/fe-core/src/main/java/com/starrocks/external/HiveMetaStoreTableUtils.java:
```diff
@@ -229,23 +233,106 @@ public static Type convertColumnType(String hiveType) throws DdlException {
+                } else {
+                    primitiveType = PrimitiveType.VARCHAR;
+                }
+                return ScalarType.createDecimalV3Type(primitiveType, precision, scale);
```
> 

done

adzfolc commented:
run starrocks_fe_unittest

adzfolc commented:
run starrocks_fe_unittest

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 43 / 137 (31.39%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|29|10.34%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|17|58|29.31%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

stephen-shelby commented:
you could add another ut to improve coverage. It's better to cover the newly written code

miomiocat commented on fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaCache.java:
```diff
@@ -320,13 +329,60 @@ public Table getTable(HiveTableName hiveTableName) {
         org.apache.hadoop.hive.metastore.api.Table hiveTable = client.getTable(hiveTableName);
-        HiveTable table =  HiveMetaStoreTableUtils.convertToSRTable(hiveTable, resourceName);
+        Table table = null;
+        if (hiveTable.getSd().getInputFormat().contains("hudi")) {
```
That's basically enough.

there are 2 ways to check table validity

1. check input format like this, refer to presto: https://github.com/prestodb/presto/blob/master/presto-hudi/src/main/java/com/facebook/presto/hudi/HudiMetadata.java#L97
2. check if basePath/.hoodie directory exists or not, refer to hudi API: https://github.com/apache/hudi/blob/master/hudi-common/src/main/java/org/apache/hudi/exception/TableNotFoundException.java#L47

stephen-shelby replied:
@miomiocat handsome,  @adzfolc I think you may use the first one.

harui7890 committed:
deal with starlet compile err (#8086)

wyb committed:
[Lake] Improve create table process (#7354)

1. Get service shard storage info from StarMgr.
2. Support shard storage info and properties when creating shards.

adzfolc committed:
1. fix array type

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 55 / 147 (37.41%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|29|10.34%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|20|56|35.71%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|11|14|78.57%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|1|1|100.00%|

wuleistarrocks committed:
[Lake] Remove redundant update_worker_info (#8113)

liuyehcf committed:
[Refactor] Remove before copy libs of be (#8137)

adzfolc committed:
1. remove unused hudi properties, it is only used in class HudiTable

adzfolc committed:
1. remove unused imports

zombee0 committed:
[SR SQL Planner Summer Camp] antlr4 for CreateDbStmt (#8133)

zombee0 committed:
[SR SQL Planner Summer Camp] antlr4 for dropDbStatement (#8185)

DorianZheng committed:
[BugFix] Fix build.sh (#8201)

We can't remove ${STARROCKS_OUTPUT}/be/lib/* after install command. Otherwise, the first cp -r xxx ${STARROCKS_OUTPUT}/be/lib/yyy command will create yyy directory as a copy of xxx while what we expect is yyy/xxx

zombee0 committed:
[SR SQL Planner Summer Camp] Antlr4 for alter database rename (#8200)

zombee0 committed:
[SR SQL Planner Summer Camp] anltr4 for showCreateDbStmt (#8189)

Signed-off-by: zombee0 <flylucas_10@163.com>

xlwh committed:
[SR SQL Planner Summer Camp] Support anltr4 for showDeleteStmt (#8197)

* anltr4 for showCreateDbStmt

* fix comments

adzfolc committed:
1. modify code style
2. refactor HudiTable

blackstar-baba committed:
Mv use insert overwrite (#8041)

Change list:

fix task framework ConnectContext NPE for schedule task.
change task schedule logic, if the start time of the task is less than the current time, the task will be scheduled immediately.
mv use insert overwrite for sync data.
mv use async task if need.
drop async task after drop mv if need.

decster committed:
[Enhancement] More clear error message in DeltaWriter (#8209)

mxdzs0612 committed:
[Feature] Implement Iceberg connector (#7647)

The first version of Iceberg connector, only supports IcebergHiveCatalog currently.

adzfolc committed:
1. fe support to refresh hudi catalog table

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=7913) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=7913) before we can accept your contribution.<br/>**33** out of **35** committers have signed the CLA.<br/><br/>:white_check_mark: HangyuanLiu<br/>:white_check_mark: wuleistarrocks<br/>:white_check_mark: mofeiatwork<br/>:white_check_mark: wyb<br/>:white_check_mark: sduzh<br/>:white_check_mark: kangkaisen<br/>:white_check_mark: liuyehcf<br/>:white_check_mark: xiaoyong-z<br/>:white_check_mark: decster<br/>:white_check_mark: DorianZheng<br/>:white_check_mark: zombee0<br/>:white_check_mark: padmejin<br/>:white_check_mark: meegoo<br/>:white_check_mark: sevev<br/>:white_check_mark: Smith-Cruise<br/>:white_check_mark: Astralidea<br/>:white_check_mark: abc982627271<br/>:white_check_mark: ZiheLiu<br/>:white_check_mark: nshangyiming<br/>:white_check_mark: gengjun-git<br/>:white_check_mark: kevincai<br/>:white_check_mark: Youngwb<br/>:white_check_mark: choury<br/>:white_check_mark: blackstar-baba<br/>:white_check_mark: miomiocat<br/>:white_check_mark: xlwh<br/>:white_check_mark: zhuxt2015<br/>:white_check_mark: rubiesvelt<br/>:white_check_mark: Linkerist<br/>:white_check_mark: chaoyli<br/>:white_check_mark: mxdzs0612<br/>:white_check_mark: adzfolc<br/>:white_check_mark: harveyyue<br/>:x: dulong41<br/>:x: mergify[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=7913) it.</sub>

adzfolc committed:
1. merge iceberg connector
2. revert MetadataMgrTest

adzfolc committed:
1. merge iceberg connector
2. revert MetadataMgrTest

wanpengfei-git commented:
#### [FE PR Coverage check]

:disappointed: **fail** : 1470 / 2474 (59.42%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowLoadWarningsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/http/rest/CancelStreamLoad.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Database.java|0|13|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowLoadStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminRepairTableStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/http/rest/GetStreamLoadState.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowFunctionsStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/TableRenameClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/master/MasterImpl.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/http/rest/QueryDumpAction.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalMergeJoinOperator.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadManager.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/mysql/privilege/UserProperty.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/transaction/TxnCommitAttachment.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DropFileStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowBackupStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/SampleStatisticsCollectJob.java|0|65|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnHistogramStatsCacheLoader.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CacheDictManager.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CancelLoadStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/common/util/KafkaUtil.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowSmallFilesStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CancelBackupStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/execution/CreateFunctionExecutor.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SelectStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/AdminCancelRepairTableStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerTraceUtil.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/load/ExportJob.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateIndexClause.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowExportStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/metric/MetricRepo.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/Explain.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/analysis/SetPassVar.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/execution/RefreshMaterializedViewExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|13|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/OperatorVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/ComputeNodeProcDir.java|0|4|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|20|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalJoinOperator.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/persist/CreateTableInfo.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowRestoreStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|31|00.00%|
|:large_blue_circle:|com/starrocks/analysis/KeysDesc.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/alter/SystemHandler.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/PropertyDeriverBase.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/Analyzer.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PartitionCommitInfo.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|0|23|00.00%|
|:large_blue_circle:|com/starrocks/transaction/InsertTxnCommitAttachment.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/common/proc/CurrentQueryInfoProvider.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/task/ExportExportingTask.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashJoinOperator.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectContext.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateFileStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowRoutineLoadTaskStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/FunctionName.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTransactionStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|0|56|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowRoutineLoadStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnector.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/http/rest/LoadAction.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/TableCollectJob.java|0|32|00.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDynamicPartitionStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/HistogramStatisticsCollectJob.java|0|30|00.00%|
|:large_blue_circle:|com/starrocks/qe/ResultReceiver.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/http/rest/StorageTypeCheckAction.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/execution/DropFunctionExecutor.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CancelExportStmt.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/master/ReportHandler.java|0|26|00.00%|
|:large_blue_circle:|com/starrocks/http/action/IndexAction.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticAutoCollector.java|1|47|02.13%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|3|67|04.48%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergHiveCatalog.java|1|12|08.33%|
|:large_blue_circle:|com/starrocks/execution/DropDbExecutor.java|1|10|10.00%|
|:large_blue_circle:|com/starrocks/analysis/TimestampArithmeticExpr.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaCache.java|3|19|15.79%|
|:large_blue_circle:|com/starrocks/transaction/GlobalTransactionMgr.java|1|4|25.00%|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|1|4|25.00%|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|3|8|37.50%|
|:large_blue_circle:|com/starrocks/external/HiveMetaStoreTableUtils.java|22|58|37.93%|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJob.java|19|50|38.00%|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|10|26|38.46%|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|4|10|40.00%|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|14|30|46.67%|
|:large_blue_circle:|com/starrocks/common/util/TimeUtils.java|3|6|50.00%|
|:large_blue_circle:|com/starrocks/analysis/SetStmt.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/http/BaseAction.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/InfoSchemaDb.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/analysis/LabelName.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java|6|12|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceClient.java|5|10|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SemiReorderRule.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/analysis/KillStmt.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|8|15|53.33%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiMetadata.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/planner/JoinNode.java|13|22|59.09%|
|:large_blue_circle:|com/starrocks/analysis/ShowDeleteStmt.java|3|5|60.00%|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|30|49|61.22%|
|:large_blue_circle:|com/starrocks/external/iceberg/IcebergUtil.java|25|40|62.50%|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|24|37|64.86%|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|12|18|66.67%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|10|15|66.67%|
|:large_blue_circle:|com/starrocks/planner/MysqlScanNode.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalNestLoopJoinOperator.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/connector/hudi/HudiConnectorFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/scheduler/MvTaskRunProcessor.java|42|59|71.19%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|51|70|72.86%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|29|39|74.36%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|53|70|75.71%|
|:large_blue_circle:|com/starrocks/planner/StreamLoadScanNode.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|14|18|77.78%|
|:large_blue_circle:|com/starrocks/planner/NestLoopJoinNode.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/journal/bdbje/BDBEnvironment.java|42|53|79.25%|
|:large_blue_circle:|com/starrocks/alter/Alter.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/catalog/StarOSAgent.java|84|104|80.77%|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|33|41|80.49%|
|:large_blue_circle:|com/starrocks/analysis/RecoverDbStmt.java|4|5|80.00%|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnector.java|13|16|81.25%|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|22|27|81.48%|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateDbAnalyzer.java|9|11|81.82%|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|5|6|83.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDbQuotaAnalyzer.java|15|18|83.33%|
|:large_blue_circle:|com/starrocks/sql/analyzer/RecoverDbAnalyzer.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|6|7|85.71%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterDatabaseRenameAnalyzer.java|12|14|85.71%|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|24|28|85.71%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|20|23|86.96%|
|:large_blue_circle:|com/starrocks/sql/analyzer/DropStmtAnalyzer.java|7|8|87.50%|
|:large_blue_circle:|com/starrocks/catalog/lake/LakeTable.java|15|17|88.24%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|51|56|91.07%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/BinaryPredicateStatisticCalculator.java|74|80|92.50%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownLimitRankingWindowRule.java|55|58|94.83%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|31|32|96.88%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|35|36|97.22%|
|:large_blue_circle:|com/starrocks/sql/optimizer/PruneShuffleColumnRule.java|98|101|97.03%|
|:large_blue_circle:|com/starrocks/metric/ResourceGroupMetricMgr.java|49|50|98.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/RequiredPropertyDeriver.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/analysis/RevokeStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskRunStatus.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleSet.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Bucket.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowAlterStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/http/rest/TableQueryPlanAction.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDataStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/planner/StreamLoadPlanner.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/cluster/Cluster.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/QueryStatement.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/join/MultiJoinNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|9|9|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateRankingWindowRule.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/planner/SortNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/DeletePlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/http/rest/TableRowCountAction.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/NestLoopJoinImplementationRule.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SetUserPropertyStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionStateListenerFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterDatabaseRename.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/mysql/privilege/Auth.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionLogApplierFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/http/rest/TableSchemaAction.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTabletStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/persist/TaskSchedule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/Histogram.java|22|22|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnStatistic.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/planner/ExchangeNode.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/analysis/GrantStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunManager.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java|14|14|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/Constants.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/TablePattern.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateRoleStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/TableProperty.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/mysql/MysqlProto.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/WorkGroupAnalyzer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/ErrorHandler.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterDatabaseQuotaStmt.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/system/Backend.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptExpressionVisitor.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowCreateDbStmt.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergConnectorFactory.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/planner/PlanFragment.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowUserPropertyStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskRun.java|12|12|100.00%|
|:large_blue_circle:|com/starrocks/external/hive/HiveMetaClient.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsEstimateUtils.java|12|12|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/Type.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropRoleStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateUserStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/OperatorType.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/connector/ConnectorMgr.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/AlterClause.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowPartitionsStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/planner/PlanNode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/util/ListUtil.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DropDbStmt.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/cluster/ClusterNamespace.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateDbStmt.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/cost/CostModel.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/common/InternalErrorCode.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/UserIdentity.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/PreAggregateTurnOnRule.java|3|3|100.00%|

Youngwb committed:
[BugFix] Derive group logical property after prune columns when need optimize CTE (#8289)
